### PR TITLE
fix: Flaky Tests in ConversationStatusLineTests 

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

We have several flaky tests in `ConversationStatusLineTests`.

Causes (Optional)

The text messages in this test have the same server time. 

Solutions

Set a different timestamp for each message

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift [hunk 1]
```diff
@@ -123,7 +123,9 @@ class ConversationStatusLineTests: CoreDataSnapshotTestCase {
         // GIVEN
         let sut = self.otherUserConversation!
         for index in 1...5 {
-            (try! sut.appendText(content: "test \(index)") as! ZMMessage).sender = self.otherUser
+            let message = try! sut.appendText(content: "test \(index)") as! ZMClientMessage
+            message.sender = self.otherUser
+            message.serverTimestamp = Date.init(timeIntervalSinceNow: Double(index))
```
Comment (netbe) [inline wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift]:
```suggestion
            message.serverTimestamp = Date(timeIntervalSinceNow: TimeInterval(index))
```

File: wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift [hunk 2]
```diff
@@ -123,7 +123,9 @@ class ConversationStatusLineTests: CoreDataSnapshotTestCase {
         // GIVEN
         let sut = self.otherUserConversation!
         for index in 1...5 {
-            (try! sut.appendText(content: "test \(index)") as! ZMMessage).sender = self.otherUser
+            let message = try! sut.appendText(content: "test \(index)") as! ZMClientMessage
```
Comment (netbe) [inline wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift]: let's make the method throwable and replace `try!` by `try`

Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift]: @netbe Done!

**Review by netbe — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

File: wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift [hunk 3]
```diff
@@ -119,11 +119,13 @@ class ConversationStatusLineTests: CoreDataSnapshotTestCase {
         XCTAssertEqual(status.string, "")
     }
 
-    func testStatusForMultipleTextMessagesInConversation() {
+    func testStatusForMultipleTextMessagesInConversation() throws {
         // GIVEN
         let sut = self.otherUserConversation!
         for index in 1...5 {
-            (try! sut.appendText(content: "test \(index)") as! ZMMessage).sender = self.otherUser
+            let message = try sut.appendText(content: "test \(index)") as! ZMClientMessage
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift]:
```suggestion
            let message = try sut.appendText(content: "test \(index)") as? ZMClientMessage
```

File: wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift [hunk 4]
```diff
@@ -141,7 +143,12 @@ class ConversationStatusLineTests: CoreDataSnapshotTestCase {
         let selfMessage = appendSelfMessage(to: sut)
 
         for index in 1...2 {
-            appendReply(to: sut, selfMessage: selfMessage, text: "reply test \(index)")
+            appendReply(
+                to: sut,
+                selfMessage: selfMessage,
+                text: "reply test \(index)",
+                timestamp: Date(timeIntervalSinceNow: -Double(2 - index))
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift]: Please check changes in my PR

Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift]: The extension is in `DataModel`:

```swift
extension ZMMessage {

     func updateTimestamp(with timeInterval: TimeInterval) {
         self.serverTimestamp = Date(timeIntervalSince1970: timeInterval)
     }

 }
 ```
 
 We can either import DataModel in this test class or move the extension to somewhere else? Any suggestions? @KaterinaWire

Comment (KaterinaWire) [reply wire-ios/Wire-iOS Tests/ConversationList/ConversationListCell/ConversationStatusLineTests.swift]: if it's just one test in the Wre-iOS project, I think you can replace it with `Date(timeIntervalSince1970: Double(index))`

**Pull request merged**